go/vt/wrangler/vdiff.go

```diff
@@ -358,10 +360,20 @@ func (df *vdiff) buildTablePlan(table *tabletmanagerdatapb.TableDefinition, quer
 	for _, pk := range table.PrimaryKeyColumns {
 		found := false
 		for i, selExpr := range targetSelect.SelectExprs {
-			colname := selExpr.(*sqlparser.AliasedExpr).Expr.(*sqlparser.ColName).Name.Lowered()
-			if pk == colname {
+			expr := selExpr.(*sqlparser.AliasedExpr).Expr
+			var colname string
+			switch ct := expr.(type) {
+			case *sqlparser.ColName:
+				colname = ct.Name.String()
+			case *sqlparser.FuncExpr: //eg. weight_string()
+				colname = ct.Name.String()
+			default:
+				log.Warningf("Unhandled type found for column in vdiff: %v(%v)", selExpr, ct)
+				colname = ""
+			}
+			if strings.EqualFold(pk, colname) {
```

Comment:
the weight_string() can't be the PK, right? shouldn't we do this only for `ColName`s?

Reply:
* I had added the e2e earlier by adding collation to a table in the existing vreplication e2e. 
* Refactored the code to isolate the logic that was changed and added unit tests. 
* I was not sure if there were FuncExprs other than weight_string() that could serve as a PK in MySQL. So just to confirm: only ColNames can be PKs, right? I updated the code to only allow ColNames